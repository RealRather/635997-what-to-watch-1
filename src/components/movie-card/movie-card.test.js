import React from 'react';
import renderer from 'react-test-renderer';
import MovieCard from './movie-card.jsx';

const mock = {
  id: `22`,
  name: `Bohemian`,
  src: `img/bohemian.jpg`
};

it(`Render MovieCard`, () => {
  const movieName = mock;
  const clickHandler = jest.fn();
  const focusHandler = jest.fn();
  const tree = renderer
    .create(<MovieCard
      movieName={movieName}
      onClick={clickHandler}
      onFocus={focusHandler}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
