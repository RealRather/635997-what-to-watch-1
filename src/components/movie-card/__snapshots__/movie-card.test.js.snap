// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MovieCard 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onFocus={[MockFunction]}
>
  <button
    className="small-movie-card__play-btn"
    onClick={[Function]}
    type="button"
  >
    Play
  </button>
  <div
    className="small-movie-card__image"
  >
    <img
      alt="Bohemian"
      height="175"
      src="img/bohemian.jpg"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
    onClick={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Bohemian
    </a>
  </h3>
</article>
`;
